### ISSUE TITLE
[flow] `declare opaque type` with two preceding block comments is emitted without `declare`

### ISSUE THREAD

**Issue by bradzacher** [prettier/prettier#13848]
**Prettier 2.7.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6AVOgOlABO1HDbPAnAEzjABsBDAJzlwgAdaBHAVyZgE8WmAFX5wkuAM4x6ASygBzANw4QAGhCsY06OOSgG9CAHcACgwQ6Utaodq8dagEb1aYANZwYAZVoBbOABlZOGQAMytxOEdnNw9PNjBZOWQpbjU4Hwc4ckpyf1p5Tlo5OAAxCHofWhhNeWQQWk4YCFUQAAsYH2oAdVbpeHF4uE9zPukANz7eOrBxexBZCPoYY2c5StDwyJAAK3EAD09E6jgARU4IeA3qCLU2ekW6kOojFpYZWC7pchhW5AAOAAMtwMES6zhYdTecEWY2Cai4FzgK1YFnq4gAtFA4Fksi1GFxpIwVkV1kgwtcthEfNJkvRUiBxEdTudLmTNmoYLQHJ9vr8kAAmDnOaTURIAYQgPlJIGhAFYWpwIoIuRZyTcQGNuABJKCUWCeMAyFgwACCus8fGOVwiAF8bUA)

```sh
--parser flow
```

**Input:**

```jsx
/**
 */
/**
 */
declare opaque type Type: string;

```

**Output:**

```jsx
/**
 */
/**
 */
opaque type Type: string;

```

**Expected behavior:**

```jsx
/**
 */
/**
 */
declare opaque type Type: string;

```

**Comment by bradzacher** [comment]
Interesting notes:

This occurs even if there are other statements are between the comments:
```jsx
type A = 1;

/**
 */
type T = 1;

/**
 */
declare opaque type Type: string;
```
[playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEMCeAHOACAgtgXmwEYBuAHSkoHoAqWy7W6y9LbAFUJIqqjoZQmLKABM4YADYBDAE44IGaQEcArjjY4OmOEmwBnGLICWUAOa8QAGhCKYx6PuSg5siAHcACnIROU0yXdpNCcbACNZaTAAazgYAGVpAFs4ABlTOGQAMwD9OHDImLj4pTBTM2QjdRs4JLC4UXFRVOlzVWkzOAAxCFkk6Rh7c2QQaVUYCGsQAAsYJMkAdWnjeH1SuHjfFeMANxW0EbB9UJBTPNkYT0izfuzc-JAAK30AD3jyyTgARVUIeDvJHkbEpZOcRllJB4phgTLAFsZRDBpsgABwABmBbjyC0iGBGMLg5x2mRsaj+cCuij8o30AFooHAGg0pvI1MZ5FcOrckDlAQ88kljJVZNUQPoPt9fv8efcbDBpGF4YjkUgAExyyLGSTlADCECS3JAhIArFNVHkOAq-LygSAduoAJJiBAJMAmDAwXBieLoT4AvIAXwDQA)

if there are multiple in a row, then only the first one is impacted:
```jsx
/**
 */
/**
 */
declare opaque type Type1: string;
declare opaque type Type2: string;
```
[playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6AVOgOlABO1HDbPAnAEzjABsBDAJzlwgAdaBHAVyZgE8WmAFX5wAjElwBnGPQCWUAOYBuClTqNmbLjxG5hAgEwTpcxSqggANCFYxZ0SclAN6EAO4AFBgkcpa1N1peR2sAI3paMABrOBgAZVoAWzgAGXk4ZAAzf0k4MIjo2Li2MHkFZBluazhE0LhySnIU2kVOWgU4ADEIekTaGDtFZBBaThgIKxAACxhE6gB1Kdl4SRK4OJ9l2QA3Zd5hsEkQkHlc+hgPCIU+rJy8kAArSQAPOLLqOABFTgh4W+pctY2PQzsNMtR3JMWKYYPNZOQYFNkAAOAAMQNcuXmERYw2hcDO2wy1i4vzgl1YvhGkgAtFA4PV6pNGFxZIxLu0bkhsgD7rlErIKvQqiBJO8vj8-ty7tYYLRQnCEUikAZZRFZNQygBhCCJLkgAkAVkmnFygnlvh5gJA224AEkoJRYHEwHIWDAAIKOuJ8D7-XIAXwDQA)

However if there is any form of a comment positioned in such a way that the the comment is attached to the second one, then it too will be impacted:
```jsx
/**
 */
/**
 */
declare opaque type Type1: string;
// impact
declare opaque type Type2: string;
```
```jsx
/**
 */
/**
 */
declare opaque type Type1: string;
declare opaque type Type2: string; // impact
```
```jsx
/**
 */
/**
 */
declare opaque type Type1: string;
declare opaque type Type2: string;
// impact
```
```jsx
/**
 */
/**
 */
declare opaque type Type1: string;
declare opaque type Type2: string;
// NOT impact as it is attached to the next type
type T = 1;
```

This pattern continues for (as far as I can tell) infinite opaque types.

**Comment by fisker** [comment]
Fixed by #13928
